### DISCUSSION TITLE
Fail2ban blocks IP filter

### DISCUSSION THREAD

**Discussion by MisTerM7** [rspamd/rspamd#4905] [General]
Hello
My question, does rspamd have its own filter to block spam IPs when they are considered as spam.
Kind regards
Michel

**Comment by fatalbanana** [comment]
It's not a generally recommendable idea.

Bad hosts will likely remain blocked by RBLs.

Hosts which send you spam might also send you mail you want & blocking at TCP-level will cause messages to pile up on remote hosts & people to become confused. If you must block, it is better to do it at the SMTP level & respond with a helpful message.

Classifications might not be a reliable indication in themselves.

Rspamd has a [reputation](https://rspamd.com/doc/modules/reputation.html) module which is intended to skew bias for hosts which are known to send mostly-spam and vice-versa.
